framework/architecture/businessunit/hierarchical-business-units.rst

```diff
@@ -0,0 +1,165 @@
+
+**hierarchical** - With the hierarchical strategy the read/write access is restricted to entities in the
+current business unit and all child business units.
+
+**self** - With the self strategy the read/write access is restricted to the current business unit.
+
+.. warning::
+    A hierarchical business unit handling only can be applied to entities which are **not** business unit dependent.
+    Entity models which are defined with **business-unit="single"** can't be used for hierarchical business units.
+
+Example
+~~~~~~~
+
+In the following picture a simple business unit structure with three levels is shown.
+
+.. figure:: resources/hierarchical_business_units_structure.png
+
+**Example 1:** Users should only have write access on users in the same business unit and on users in child business
+units.
+
+This means a user in the business unit `Switzerland` should have write access on all users while a user in the business
+unit `Zurich` should have write access on users in the business units `Zurich`, `Zurich 1` and `Zurich 2`.
+
+Solution: use the `hierarchical` strategy for the write access for the entity model `User`.
+
+**Example 2:** User should only have read access on users in the same business unit and on users in child business units
+but write access only on users in the same business unit.
+
+This means a user in the business unit `Switzerland` should have read access on all users but only has write access on
+users in the business unit `Switzerland`. A user in the business unit `Zurich` should have read access on users in the
+business units `Zurich`, `Zurich 1` and `Zurich 2` but only has write access on users in `Zurich`.
+
+Solution: use the `hierarchical` strategy for the read access and the `self` strategy for the write access for the entity
+model `User`.
+
+How to add Hierarchical Structures
+----------------------------------
+
+To add a hierarchical business unit handling the following steps need to be done
+
+1) Configure a hierarchical business unit structure
+2) Add relations from the desired entity model to the business unit
+3) Add a contribution in the ``hivemodule.xml`` which specifies the strategy
+
+Assume we want to apply the hierarchical business unit handling to the entity model `Address`.
+
+Configure a hierarchical business unit structure
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Hierarchical business unit structures can easyily be configured by setting the relation ``relParent`` and ``relChildren``
+correctly on the business unit entities.
+
+Add relation ``relHierarchy_business_unit``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+It is necessary to add relation to the business unit for each entity model which should be applied to the hierarchical
+business unit handling. It is important to name the relation something different than `relBusiness_unit`. Otherwise
+the `counter fragment` creates a counter for each business unit for that entity model which is not desired.
```

Comment:
I think there is a comma needed before "which" in this case. Without the comma "which" refers to "entity model".

Reply:
it should refer to the entity mode :) the entity model is applied to the hierarchical business unit handling.